### PR TITLE
Arc ui image

### PR DESCRIPTION
**Pull request by mattshoobs**
Adding default image and image upload and display functionality.
Refactoring app structure into ui folder for master branch.
Jira: ARC-90 https://tim-kaufenberg.atlassian.net/browse/ARC-90
Jira: ARC-89 https://tim-kaufenberg.atlassian.net/browse/ARC-89

### CODE REVIEW

**Comment by coughenberg** [comment]
Like the comment in #11 I cannot review this story since I do not know the acceptance criteria on the Jira story

**Comment by mattshoobs** [comment]
We are not able to modify completed jira stories to add acceptance criteria. The stories linked above describe the functionality we are looking for.

**Comment by coughenberg** [comment]
But they are not complete and are still in the review process. There is also a review button on the story progress options, per your request when we started the arc project, so this should not be an issue. Please move it off of done and onto review since it is not done.
This story has no description and acceptance criteria, so it is too generic of a story as it can be applied to a lot of different scenarios https://tim-kaufenberg.atlassian.net/browse/ARC-90
How does this impact the user? How do we know when it is done? Where is this uploaded image going in the app?
If you need help making user stories, websites like [this](https://www.freecodecamp.org/news/how-and-why-to-write-great-user-stories-f5a110668246/) are a good place to start

File: ui/package.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    "electron": "9.1.1",
+    "eslint": "^6.8.0",
+    "eslint-plugin-import": "^2.20.0",
+    "file-loader": "^6.0.0",
+    "fork-ts-checker-webpack-plugin": "^3.1.1",
+    "node-loader": "^0.6.0",
+    "style-loader": "^0.23.1",
+    "ts-loader": "^6.2.1",
+    "typescript": "^3.7.0"
+  },
```
Comment (coughenberg) [inline ui/package.json:82]: It looks like these lines were added when moving the package.json over, was this meant to happen?

File: ui/src/App.tsx [hunk 2]
```diff
@@ -150,7 +166,10 @@ const App = () => {
           <TextField style={{margin:"10px"}} id="outlined-basic" label="Armspan" variant="outlined" />
           <Divider />
           <Typography style={{margin:"10px"}}>2. Upload the route image</Typography>
-          <TextField style={{margin:"10px"}} id="outlined-basic" label="File Upload" variant="outlined" />
+          <Button style={{margin:"10px"}} variant="contained" color="secondary" onClick={() => { handleOpenImage()}}> Upload File <input
+    type="file"
+    style={{ display: "none" }}
+  /></Button>
```
Comment (coughenberg) [inline ui/src/App.tsx]: This needs to be reformatted. 

1. Line is too long so move "Upload File" to next line.
2. We shouldn't start a new tag a the end of a line and not end the first tag there, it is misleading. For this we can just have input on the next line and probably have all of input on one line.
3. Move the "/>" on the last line up so the end carrot is not floating

File: ui/src/index.ts [hunk 3]
```diff
@@ -11,13 +11,16 @@ const createWindow = () => {
...
+      nodeIntegration: true
+    }
   });
 
   // and load the index.html of the app.
   mainWindow.loadURL(MAIN_WINDOW_WEBPACK_ENTRY);
 
   // Open the DevTools.
-  mainWindow.webContents.openDevTools();
+  //mainWindow.webContents.openDevTools();
```
Comment (coughenberg) [inline ui/src/index.ts]: Get rid of commented out code, it shouldn't be in production

File: ui/package.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
+{
```
Comment (coughenberg) [inline ui/package.json:1]: For visibility https://github.com/ARC-Summer-2020/AutoRockClimber/pull/11#discussion_r479696332

File: package-lock.json [hunk 5]
```diff
@@ -1,9327 +0,0 @@
-{
-  "name": "arc-ui-app",
```
Comment (coughenberg) [inline package-lock.json]: Package-lock has been deleted but doesn't appear to have been brought back. Please readd package-lock

**Review by coughenberg — CHANGES_REQUESTED** [review]

File: ui/package.json (hunk 1 above)
Comment (mattshoobs) [reply ui/package.json:82]: yes, these dependencies are needed for the app to function, they were somehow removed

File: package-lock.json (hunk 5 above)
Comment (mattshoobs) [reply package-lock.json]: Somehow it was in the .gitignore, removed and added to repo

File: ui/src/index.ts (hunk 3 above)
Comment (mattshoobs) [reply ui/src/index.ts]: Fixed in latest commit

File: ui/src/App.tsx (hunk 2 above)
Comment (mattshoobs) [reply ui/src/App.tsx]: Fixed in latest commit

**Review by skitscha — APPROVED** [review]

File: ui/src/App.tsx [hunk 6]
```diff
@@ -166,10 +162,10 @@ const App = () => {
           <TextField style={{margin:"10px"}} id="outlined-basic" label="Armspan" variant="outlined" />
           <Divider />
           <Typography style={{margin:"10px"}}>2. Upload the route image</Typography>
-          <Button style={{margin:"10px"}} variant="contained" color="secondary" onClick={() => { handleOpenImage()}}> Upload File <input
-    type="file"
-    style={{ display: "none" }}
-  /></Button>
+          <Button style={{margin:"10px"}} variant="contained" color="secondary" 
+          onClick={() => {handleOpenImage()}}> 
+            Upload File <input type="file" style={{ display: "none" }}/>
```
Comment (coughenberg) [inline ui/src/App.tsx]: If you are starting a new tag, it is good practice for it to be on the left-hand side so it is easy to identify where it is when editing code. Move the <input> tag and info to the next line

**Review by coughenberg — CHANGES_REQUESTED** [review]

**Review by coughenberg — APPROVED** [review]

**Pull request merged**